src/features/events/components/EventOverviewCard/EventOverviewCard.tsx

```diff
@@ -0,0 +1,186 @@
+import EditIcon from '@mui/icons-material/Edit';
+import LocationOnIcon from '@mui/icons-material/LocationOn';
+import OpenInNewIcon from '@mui/icons-material/OpenInNew';
+import {
+  Box,
+  Button,
+  Card,
+  ClickAwayListener,
+  Divider,
+  Link,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { FC, useState } from 'react';
+
+import EventDataModel from 'features/events/models/EventDataModel';
+import { getWorkingUrl } from 'features/events/utils/getWorkingUrl';
+import messageIds from 'features/events/l10n/messageIds';
+import theme from 'theme';
+import UseEditEventOverviewCard from './useEditEventOverviewCard';
+import { useMessages } from 'core/i18n';
+import { ZetkinEvent } from 'utils/types/zetkin';
+import ZUIPreviewableInput from 'zui/ZUIPreviewableInput';
+
+type EventOverviewCardProps = {
+  editable: boolean;
+  element: ZetkinEvent;
+  model: EventDataModel;
+  onEditModeEnter: () => void;
+  onEditModeExit: () => void;
+};
+
+const EventOverviewCard: FC<EventOverviewCardProps> = ({
+  editable,
+  element,
+  model,
+  onEditModeEnter,
+  onEditModeExit,
+}) => {
+  const messages = useMessages(messageIds);
+  const [startDate, setStartDate] = useState(element.start_time);
+  const [endDate, setEndDate] = useState(element.end_time);
+  const [link, setLink] = useState(element.url);
+  const [infoText, setInfoText] = useState(element.info_text);
+
+  const { clickAwayProps, containerProps, previewableProps } =
+    UseEditEventOverviewCard({
+      editable,
+      onEditModeEnter,
+      onEditModeExit,
```

Comment:
Instead of passing these from the outside, you should be able to just do this:

```tsx
onEditModeEnter: () => setEditable(true),
onEditModeExit: () => setEditable(false),
```

And of course add a `const [editable, setEditable] = useState(false)` state to the component. These changes would mean that the `editable`, `onEditModeExit` and `onEditModeEnter` props are no longer needed, and all of it is instead handled internally in the component.